homeassistant/components/camera/__init__.py

```diff
@@ -806,7 +831,29 @@ def async_write_ha_state(self) -> None:
         ):
             self.__supports_stream = supports_stream
             self._invalidate_camera_capabilities_cache()
-            self.hass.async_create_task(self.async_refresh_providers())
+
+            async def refresh() -> None:
+                await self.async_refresh_providers()
+                await self.async_create_go2rtc_client_if_needed()
```

Comment:
why is this call needed here, if it is already called on async_internal_added_to_hass ?